src/shared/income-filter/test/income-filter.component.spec.ts

```diff
@@ -1,60 +1,133 @@
+
+const defaultStreetSettings = {
+  lowIncome: 30,
+  highIncome: 60
+};
 
 describe('IncomeFilterComponent', () => {
-    let fixture: ComponentFixture<IncomeFilterComponent>;
-    let component: IncomeFilterComponent;
-
-    beforeEach(async(() => {
-        TestBed.configureTestingModule({
-            imports: [
-                TranslateModule,
-                StoreModule.forRoot({})
-            ],
-            declarations: [
-                IncomeFilterComponent,
-                StreetFilterComponent
-            ],
-            providers: [
-                MathService,
-                StreetFilterDrawService,
-                { provide: TranslateService, useClass: TranslateServiceMock },
-                { provide: TranslateLoader, useClass: TranslateLoaderMock },
-                { provide: TranslateParser, useClass: TranslateParserMock }
-            ]
-        });
-
-        fixture = TestBed.createComponent(IncomeFilterComponent);
-        component = fixture.componentInstance;
-    }));
-
-    it('ngAfterViewInit()', () => {
-        component.ngAfterViewInit();
-
-        expect(component.streetSettingsStateSubscription).toBeDefined();
-
-        spyOn(component.streetSettingsStateSubscription, 'unsubscribe');
-
-        component.ngOnDestroy();
-
-        expect(component.streetSettingsStateSubscription.unsubscribe).toHaveBeenCalled();
+  let fixture: ComponentFixture<IncomeFilterComponent>;
+  let component: IncomeFilterComponent;
+  let store: StoreMock;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        TranslateModule.forRoot({
+          provide: TranslateLoader,
+          useFactory: (createTranslateLoader),
+          deps: [Http]
+        })
+      ],
+      schemas: [NO_ERRORS_SCHEMA],
+      declarations: [
+        IncomeFilterComponent
+      ],
+      providers: [
+        MathService,
+        {provide: StreetFilterDrawService, useValue: {}},
+        {provide: Store, useClass: StoreMock}
+      ]
     });
 
-    it('initData()', () => {
-        component.ngAfterViewInit();
+    fixture = TestBed.createComponent(IncomeFilterComponent);
+    store = TestBed.get(Store);
+
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should unsubscribe from streetSettingsStateSubscription on destroy', () => {
+    expect(component.streetSettingsStateSubscription).toBeDefined();
+
+    spyOn(component.streetSettingsStateSubscription, 'unsubscribe');
+    component.ngOnDestroy();
+
+    expect(component.streetSettingsStateSubscription.unsubscribe).toHaveBeenCalled();
+  });
+
+  it('call initData()', () => {
+    component.initData();
+  });
+
+  it('applyFilter should update range settings and close the range', () => {
+    const expectedRange: any = {
+      lowIncome: 1,
+      highIncome: 2
+    };
+    let selectedRange;
+    const okBtn = fixture.debugElement.query(By.css('.ok-button')).nativeElement;
```

Comment:
Make variable `element = fixture.debugElement` in beforEach.
And use in tests const okBtn = element.query(By.css('.ok-button')).nativeElement;